repository-data/webfiles/src/main/resources/site/freemarker/common/apispecification.ftl

```diff
@@ -1,19 +1,41 @@
 <#ftl output_format="HTML">
 
+<#include "../include/imports.ftl">
 <#include "macro/documentHeader.ftl">
 
 <#-- @ftlvariable name="document" type="uk.nhs.digital.website.beans.ApiSpecification" -->
+<#-- @ftlvariable name="hstRequestContext" type="org.hippoecm.hst.core.request.HstRequestContext" -->
 
 <#if document?? >
-    <article class="article article--apispecification" itemscope>
-        <@documentHeader document 'general'></@documentHeader>
 
-        <div class="grid-wrapper grid-wrapper--article">
-            <div class="grid-row">
+    <#assign isTryItNow = hstRequestContext.servletRequest.parameterMap?keys?seq_contains("tryitnow")>
+    <#if isTryItNow >
 
-                ${document.html?no_esc}
+        <#include "api-try-it-now.ftl">
 
+    <#else>
+
+        <article class="article article--apispecification" itemscope>
+            <@documentHeader document 'general'></@documentHeader>
+
+            <style type="text/css">
+                .ctabtn--nhs-digital-button--try-it-now { float: right; }
+            </style>
+
+            <div class="grid-wrapper grid-wrapper--article">
+                <div class="grid-row">
+
+                    ${document.html?no_esc}
```

Comment:
Re: conversation on supplying a value to a mustache template's href from this later context, you could add a placeholder in mustache and then do a string replace here. For anything more complicated, such as replacing more than one value, you could feed `document.html` to an implementation of a Freemarker TemplateMethodModelEx. Then print the returned object from your TemplateMethodModelEx.

Reply:
THAT is a cool idea. Thanks, Geoff.

Admittedly, it means that the entire (substantial) `document.html` payload will have to be processed on each request but there are many documents with large-is payload that we process with Freemarker anyway, so it shouldn't really be a concern.